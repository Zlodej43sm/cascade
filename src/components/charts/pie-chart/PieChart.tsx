import React from 'react';

import Chart, { ChartProps } from 'lib/components/charts/chart/Chart';

function PieChart(props: ChartProps) {
  return (
    <Chart {...props} />
  );
}

export default PieChart;
