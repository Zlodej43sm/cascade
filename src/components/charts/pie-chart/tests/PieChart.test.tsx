import { render } from '@testing-library/react';
import Highcharts from 'highcharts';

import { chartArguments } from '../__mocks__/PieChart.mock';
import PieChart from '../PieChart';

const chartImplementation = {
  reflow: jest.fn(),
  destroy: jest.fn(),
  update: jest.fn(),
};

jest.mock('highcharts', () => ({
  chart: jest.fn().mockImplementation(() => chartImplementation),
}));

describe('TableChart', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<PieChart highcharts={Highcharts} options={chartArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
