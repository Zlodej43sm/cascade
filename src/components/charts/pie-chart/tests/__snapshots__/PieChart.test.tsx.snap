// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableChart Should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="chart-wrapper"
  />
</DocumentFragment>
`;
